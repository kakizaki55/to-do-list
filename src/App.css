.todo-list {
  border: black, solid, 2px;
}
