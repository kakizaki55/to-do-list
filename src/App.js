import './App.css';
import { React, useState } from 'react';
import { BrowserRouter } from 'react-router-dom';
import Header from './components/Header/Header';
import Auth from './views/Auth/Auth';
import { getUser, logout } from './services/fetchUser';
import TodoList from './views/TodoList/TodoList';

function App() {
  const [currentUser, setCurrentUser] = useState(getUser());

  const logoutUser = async () => {
    await logout();
    setCurrentUser(null);
  };

  return (
    <div className="App">
      <BrowserRouter>
        <Header></Header>

        {currentUser ? (
          <>
            <div>this is the logged in page</div>
            <TodoList></TodoList>
            <button onClick={logoutUser}>logout</button>
          </>
        ) : (
          <Auth setCurrentUser={setCurrentUser}></Auth>
        )}
      </BrowserRouter>
    </div>
  );
}

export default App;
