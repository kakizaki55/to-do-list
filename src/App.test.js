import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', async () => {
  const container = render(<App />);
  await screen.findByText('Todo Manager');
  expect(container).toMatchSnapshot();
});
