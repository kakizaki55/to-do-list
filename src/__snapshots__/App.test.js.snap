// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header>
          <h1>
            Todo Manager
          </h1>
          <div
            class="logout-header"
          >
            <h3
              class="logout-button"
            >
              logout
            </h3>
          </div>
        </header>
        <p />
        <div>
          <h1>
            Login Page:
          </h1>
          <div>
            <h5
              class=""
            >
              Sign Up
            </h5>
          </div>
          <div>
            <h5
              class="active"
            >
              Sign In
            </h5>
          </div>
          <form>
            <label>
              Email:
            </label>
            <input
              placeholder="Email"
              type="email"
              value=""
            />
            <label>
              Password:
            </label>
            <input
              placeholder="Password"
              type="password"
              value=""
            />
            <button>
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header>
        <h1>
          Todo Manager
        </h1>
        <div
          class="logout-header"
        >
          <h3
            class="logout-button"
          >
            logout
          </h3>
        </div>
      </header>
      <p />
      <div>
        <h1>
          Login Page:
        </h1>
        <div>
          <h5
            class=""
          >
            Sign Up
          </h5>
        </div>
        <div>
          <h5
            class="active"
          >
            Sign In
          </h5>
        </div>
        <form>
          <label>
            Email:
          </label>
          <input
            placeholder="Email"
            type="email"
            value=""
          />
          <label>
            Password:
          </label>
          <input
            placeholder="Password"
            type="password"
            value=""
          />
          <button>
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
